src/plugins/memdump/memdump.cpp

```diff
@@ -0,0 +1,252 @@
+ * and redistribute Covered Software under those terms instead of these.   *
+ *                                                                         *
+ * Source is provided to this software because we believe users have a     *
+ * right to know exactly what a program is going to do before they run it. *
+ * This also allows you to audit the software for security holes.          *
+ *                                                                         *
+ * Source code also allows you to port DRAKVUF to new platforms, fix bugs, *
+ * and add new features.  You are highly encouraged to submit your changes *
+ * on https://github.com/tklengyel/drakvuf, or by other methods.           *
+ * By sending these changes, it is understood (unless you specify          *
+ * otherwise) that you are offering unlimited, non-exclusive right to      *
+ * reuse, modify, and relicense the code.  DRAKVUF will always be          *
+ * available Open Source, but this is important because the inability to   *
+ * relicense code has caused devastating problems for other Free Software  *
+ * projects (such as KDE and NASM).                                        *
+ * To specify special license conditions of your contributions, just say   *
+ * so when you send them.                                                  *
+ *                                                                         *
+ * This program is distributed in the hope that it will be useful, but     *
+ * WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+ * license file for more details (it's in a COPYING file included with     *
+ * DRAKVUF, and also available from                                        *
+ * https://github.com/tklengyel/drakvuf/COPYING)                           *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <config.h>
+#include <glib.h>
+#include <inttypes.h>
+#include <libvmi/libvmi.h>
+#include <assert.h>
+#include "memdump.h"
+
+static bool dump_memory_region(memdump* plugin, vmi_instance_t vmi, access_context_t* ctx, size_t num_pages, vmi_pid_t pid)
+{
+    char* file = nullptr;
+    void** access_ptrs = nullptr;
+    FILE* fp = nullptr;
+    bool ret = false;
+
+    plugin->memdump_counter++;
+
+    if ( asprintf(&file, "%s/%d-0x%llx-%04d.dmp", plugin->memdump_dir, pid, (unsigned long long)ctx->addr, plugin->memdump_counter) < 0 )
+    {
+        PRINT_DEBUG("[MEMDUMP] Failed asprintf\n");
+        goto done;
+    }
+
+    access_ptrs = (void**)g_malloc(num_pages * sizeof(void*));
+
+    if (VMI_SUCCESS != vmi_mmap_guest(vmi, ctx, num_pages, access_ptrs))
+    {
+        PRINT_DEBUG("[MEMDUMP] Failed mmap guest\n");
+        goto done;
+    }
+
+    printf("[MEMDUMP] Writing dump with %d pages to: %s\n", (int)num_pages, file);
```

Comment:
This should either be a `PRINT_DEBUG` or have all the other output options supported too